### ISSUE TITLE
Shouldn't CPM objects be injected back into Autoware?

### ISSUE THREAD

**Issue by MarioAlmelaUMH** [tlab-wide/AutowareV2X#42]
Hello, I have been reading the documentation and looking at the code files, and I have found a topic I don't quite understand. I see in the v2x_node file that AutowareV2X publishes the received CPM objects in a topic called "/v2x/cpm/objects". That topic doesn't appear in Autoware documentation (https://app.diagrams.net/?lightbox=1#Uhttps%3A%2F%2Fautowarefoundation.github.io%2Fautoware-documentation%2Flatest%2Fdesign%2Fautoware-architecture%2Fnode-diagram%2Foverall-node-diagram-autoware-universe.drawio.svg) nor when using ros2 topic list command while using Autoware, so I assume this is a new topic created for the AutowareV2X project. 

My point is, since no node in Autoware subscribes to this topic, how does an AV (the ego-vehicle for instance) react or know about these objects? Shouldn't CPM received objects be injected back into Autoware's Perception Module so vehicles can react to them? Or maybe am I missing something?

Thank you in advance for your attention.

**Comment by yuasabe** [comment]
Hello @MarioAlmelaUMH 
Thank you for reaching out. Yes, the "/v2x/cpm/objects" topic is one that is prepared for this project and Autoware itself does not read from this topic yet. As per my understanding, there is no concrete interface yet for V2X applications with Autoware, so designing the necessary topics, messages, modules, etc. is still an ongoing task.

However, a simple way to inject the CPM received objects into Autoware (for R&D purposes), is to publish the objects onto the "/perception/object_recognition/objects" topic. This way, the CPM shared objects will be treated like any other locally percevied object, and Autoware's planner will behave accordingly. For some experiments I have conducted, I have used CPM shared objects in this manner. (Video of blindspot scenario: https://youtu.be/57fx3-gUNxU).

I completely agree with you on the point that we still need an agreed method of how V2X-shared information is fed into Autoware. For this, I hope we can further discuss with other Autoware developers and the AWF community.

This project is still in its early stages, but I hope it will give insight into how we can incorporate V2X technologies into Autoware to enable more advanced cooperative schemes.
